codegen/src/main/java/software/amazon/awssdk/codegen/poet/waiters/BaseWaiterClassSpec.java

```diff
@@ -488,6 +493,19 @@ private CodeBlock acceptor(Acceptor acceptor) {
         return result.build();
     }
 
+    private CodeBlock.Builder booleanValueErrorBlock(Acceptor acceptor, Boolean expectedBoolean) {
+        CodeBlock.Builder codeBlock = CodeBlock.builder();
+        if (Boolean.FALSE.equals(expectedBoolean)) {
+            codeBlock.add("OnResponseAcceptor(");
+            codeBlock.add(trueForAllResponse());
+        } else {
+            codeBlock.add("OnExceptionAcceptor(");
+            codeBlock.add("error -> errorCode(error) " + (expectedBoolean ? "!=" : "==") + " null");
```

Comment:
exepectedBoolean is also true in this else block, right?
`codeBlock.add("error -> errorCode(error) != null");`